doc/vxlan/EVPN/EVPN_VxLAN_Multihoming.md

```diff
@@ -0,0 +1,1265 @@
+- FdbOrch is extended to process new field "ifname" from VXLAN_FDB_TABLE entries.
+- FdbOrch is extended to process new field "type" for possible local/remote from VXLAN_FDB_TABLE entries.
+- FdbOrch is store and update MAC entries per ES associated PortChannel. This is to update the MAC entries when the associated PortChannel interface is going up/down. 
+- FdbOrch is interact with EvpnMhOrch to check if the given PortChanenl is associated with ES or not, fetch the L2-NHID associated with PortChannel, etc.
+- FdbOrch is extended to process FDB entries on local po down and program then pointing to backup nexthop group using EVPN_ES_BACKUP_NHG_TABLE.
+
+Fdborch to process the local po down and up in a similar fashion as for MCLAG. 
+
+### 3.3.8 L2nhgorch changes
+- New Orchagent class being introduced.
+- Subscription to different tables
+  - APP DB - L2_NEXTHOP_GROUP, L2_NEXTHOP_GROUP_MEMBER
+  - CONFIG DB - EVPN_ES_INTERFACE.
+- Subscription to L2_NEXTHOP_GROUP
+  - Creates/deletes the L2_ECMP_GROUP ASIC DB table. 
+  - Interacts with PortsOrch to create a bridge-port object of type L2_ECMP_GROUP.
+- Subscription to L2_NEXTHOP_GROUP_MEMBER
+  - Creates/deletes the L2_ECMP_GROUP_MEMBER ASIC DB tables entries on the L2_NEXTHOP_GROUP APP DB table updates.
+  - Interacts with VxlanOrch to update the refcnt per remote VTEP.
+- Subscription to EVPN_ESI CONFIG_DB table 
+  - Informs Fdborch of the ESI to PortChannel mapping.
+  - Maintain a local cache of the mapping.
+- Subscription to LAG oper-status updates from PortsOrch
+- API 
+  - L2 NHID to Bridgeport mapping.
+
+### 3.3.9 EvpnMhOrch
+A new EvpnMhOrch class will be introduced to perform following activities, 
+
+1. Subscribe to EVPN_MH_GLOBAL config table and set the SAI switch attribute. 
+2. Subscribe to EVPN_ES_INTERFACE config table updates and maintain internal cache.
+3. Provide set of API to check if the given PortChannel is ES associated or not. 
+4. Subscribe to LAG_TABLE oper status changes and trigger MAC update requests to FdbOrch.
+
+## 3.4 SAI
+
+### 3.4.1 New SAI Objects 
+No new SAI objects are introduced as part of this design.
+
+
+### 3.4.2 Changes to Existing SAI Objects
+
+- **NEXT_HOP_TYPE_L2**
+  - The nexthop type LAYER_2 is to be used for L2 ECMP nexthop group members.
+```
+/**
+ * @brief Next hop type
+ */
+typedef enum _sai_next_hop_type_t
+{
+   ...
+    /** Next hop group is for layer 2 bridging */
+    SAI_NEXT_HOP_TYPE_LAYER_2,
+} sai_next_hop_type_t;
+```
+
+- **NEXT_HOP_GROUP_TYPE_L2**
+  - The nexthop group type LAYER_2 is to be used for MAC/FDB entries pointing to L2 ECMP nexthop group.
+```
+/**
+ * @brief Next hop group type
+ */
+typedef enum _sai_next_hop_group_type_t
+{
+...
+    /** Next hop group is for bridge port */
+    SAI_NEXT_HOP_GROUP_TYPE_LAYER_2,
+
+    /* Other types of next hop group to be defined in the future, e.g., WCMP */
+} sai_next_hop_group_type_t;
+```
+
```

Comment:
List the supported sai_object_id_t that are applicable for new nexthop group type L2 in sai_next_hop_group_member_attr_t object member attribute - SAI_NEXT_HOP_GROUP_MEMBER_ATTR_NEXT_HOP_ID,